Functions/Quantification/xASL_quant_External.m

```diff
@@ -0,0 +1,706 @@
+				%fprintf(FIDoptionFile, '--repeats=1\n');
+			case 'vaby'
+				% Printing the values in the commandline (PLD=plds, LD=taus)
+				for iPLD = 1:length(PLDs)
+					if iPLD == 1
+						ExternalOptions = [ExternalOptions ' --plds=' sprintf('%.2f', PLDs(iPLD))];
+					else
+						ExternalOptions = [ExternalOptions ',' sprintf('%.2f', PLDs(iPLD))];
+					end
+				end
+
+				for iTE = 1:length(TEs) %So for each volume, we print a TE value
+					if iTE == 1
+						ExternalOptions = [ExternalOptions ' --tes=' sprintf('%.3f', TEs(iTE))];
+					else
+						ExternalOptions = [ExternalOptions ',' sprintf('%.3f', TEs(iTE))];
+					end
+				end
+			case 'basil'
+				% Specify that we run the PCASL/CASL model
+				fprintf(FIDoptionFile, '--casl\n');
+				fprintf('BASIL: (P)CASL model\n');
+
+				% For BASIL, PLDs are specified
+				if bQuantifyMultiPLD
+					for iPLD = 1:length(PLDs)
+						fprintf(FIDoptionFile, '--pld%d=%.2f\n', iPLD, PLDs(iPLD));
+					end
+				else
+					fprintf(FIDoptionFile, '--pld=%.2f\n', PLDs(1));
+				end
+		end
+
+		switch (lower(strQuantificationType))
+			case {'fabber','basil'}
+				% Print labeling durations
+				if bQuantifyMultiPLD
+					for iLabDurs = 1:length(LabDurs)
+						fprintf(FIDoptionFile, '--tau%d=%.2f\n', iLabDurs, LabDurs(iLabDurs));
+					end
+				else
+					fprintf(FIDoptionFile, '--tau=%.2f\n', LabDurs(1));
+				end
+			case 'vaby'
+				for iLD = 1:length(LabDurs) 
+					if iLD == 1
+						ExternalOptions = [ExternalOptions ' --taus=' sprintf('%.2f', LabDurs(iLD))];
+					else
+						ExternalOptions = [ExternalOptions ',' sprintf('%.2f', LabDurs(iLD))];
+					end
+				end
+		end
+end
+
+if ~isempty(regexpi(strQuantificationType, 'basil', 'once'))
+	% Act as if we do not have repeats
+	%fprintf(FIDoptionFile, '--repeats=%i\n', size(PWI, 4)/PLDAmount);
+	fprintf(FIDoptionFile, '--repeats=1\n');
+
+	% Slice-timing
+	fprintf(FIDoptionFile, '--slicedt=%f\n', x.Q.BasilSliceReadoutTime/1000);
+
+	if isfield(x.Q,'LookLocker') && x.Q.LookLocker
+		if isfield(x.Q,'FlipAngle')
+			if length(unique(x.Q.FlipAngle))>1
+				warning('Look-Locker quantification with multiple flip angles, e.g. QUASAR, is not implemented yet');
+			end
+			fprintf(option_file, '--FA=%f\n', x.Q.FlipAngle(1));
+			fprintf('BASIL: Flip angle for Look-Locker readout: %f\n', x.Q.FlipAngle(1));
+		else
+			warning('BASIL: Unknown flip angle for Look-Locker\n');
+		end
+	end
+end
+
+%% 4. Model fiting parameters
+if ~isempty(regexpi(strQuantificationType, 'basil', 'once'))
+	switch lower(x.Q.LabelingType)
+		case 'pasl'
+			% Default initial ATT for PASL is 0.7
+			fprintf(FIDoptionFile, '--bat=0.7\n');
+		case {'pcasl','casl'}
+			% Default initial ATT for PCASL is 1.3
+			fprintf(FIDoptionFile, '--bat=1.3\n');
+	end
+
+	if bQuantifyMultiPLD
+		% Multi-PLD or Time Encoded data allows to fit arrival times
+		fprintf(FIDoptionFile, '--batsd=%f\n', x.modules.asl.ATTSDBASIL);
+	end
+end
+
+%% 5. Extra BASIL fitting options
```

Comment:
Shouldn't we combine this with 4 as both are BASIL-specific?

Reply:
Sure. It was basic and advanced parameters. But since we now have also VABY and FABBER then indeed better to merge. Done in a 385a0d4d